packages/deployment/test/test_kes.js

```diff
@@ -0,0 +1,167 @@
+  t.is(result, '${FunctionNameLambdaAliasnotarealhash}');
+});
+
+test('lookupLambdaReference returns the original ', (t) => {
+  const resourceString = '${FunctionNameLambdaFunction.Arn}';
+  const kes = setupKesForLookupLambdaReference(t.context.kes, 'FunctionName', null);
+  const result = kes.lookupLambdaReference(resourceString);
+  t.is(result, resourceString);
+});
+
+test('lookupLambdaReference throws error on invalid configuration', (t) => {
+  const resourceString = '${SomeOtherConfigurationString.foobar}';
+  const kes = setupKesForLookupLambdaReference(t.context.kes, 'FunctionName', null);
+  const error = t.throws(() => {
+    kes.lookupLambdaReference(resourceString);
+  }, Error);
+  t.is(error.message, `Invalid stateObjectResource: ${resourceString}`);
+});
+
+test('injectWorkflowLambdaAliases updates the correct resources', (t) => {
+  const kes = t.context.kes;
+
+  kes.config.lambdas = {
+    TestLambda: { hash: 'notarealhash' },
+    TestUnversionedLambda: { hash: null }
+  };
+
+  kes.config.stepFunctions = {
+    TestStepFunction: {
+      States: {
+        1: { Type: 'Task', Resource: '${TestLambdaLambdaFunction.Arn}' },
+        2: { Type: 'Task', Resource: '${TestUnversionedLambdaLambdaFunction.Arn}' },
+        3: { Type: 'Task', Resource: '${SomethingElse.Arn}'}
+      }
+    },
+    TestStepFunction2: {
+      States: {
+        1: { Type: 'Task', Resource: '${TestLambdaLambdaFunction.Arn}' }
+      }
+    }
+  };
+
+  const expected = {
+    TestStepFunction: {
+      States: {
+        1: { Type: 'Task', Resource: '${TestLambdaLambdaAliasnotarealhash}' },
+        2: { Type: 'Task', Resource: '${TestUnversionedLambdaLambdaFunction.Arn}' },
+        3: { Type: 'Task', Resource: '${SomethingElse.Arn}'}
+      }
+    },
+    TestStepFunction2: {
+      States: {
+        1: { Type: 'Task', Resource: '${TestLambdaLambdaAliasnotarealhash}' }
+      }
+    }
+  };
+
+  kes.injectWorkflowLambdaAliases();
+  t.deepEqual(expected, kes.config.stepFunctions);
+});
+
+
+test('injectOldWorkflowLambdaAliases adds oldLambdas to configuration object', async (t) => {
+  const kes = t.context.kes;
+  const nameArray = ['LambdaName-12345', 'LambdaName-abcde', 'SecondLambdaName-67890'];
+  kes.getRetainedLambdaAliasNames = () => (Promise.resolve(nameArray));
+
+  const expected = {
+    LambdaName: { hashes: ['12345', 'abcde'] },
+    SecondLambdaName: { hashes: ['67890'] }
+  };
+
+  await kes.injectOldWorkflowLambdaAliases();
+  t.deepEqual(expected, kes.config.oldLambdas);
+});
+
+
+test('parseAliasName parses the alias name', (t) => {
+  const expected = { name: 'functionName', hash: 'hashValue' };
+  const actual = t.context.kes.parseAliasName('functionName-hashValue');
+  t.deepEqual(expected, actual);
+});
+
+test('parseAliasName returns null hash if hash missing', (t) => {
+  const expected = { name: 'functionName', hash: null };
+  const actual = t.context.kes.parseAliasName('functionName-');
```

Comment:
Why would aliasName have a dash on the end here?  I guess would we ever have this case where there was no hash in the name?

Reply:
In current business logic, we shouldn't.   In previous iterations/attempts we did.  I think the behavior/test is probably still valuable against the method as it's still a viable future use of the helper to have hashless functions that need aliases (to pack version number).